angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.ts

```diff
@@ -405,34 +378,30 @@ export class MinesCollectionsAddEditComponent implements OnInit, OnDestroy {
     if (this.isEditing) {
       collection['_id'] = this.collection._id;
 
-      this.factoryService.editCollection(collection).subscribe(async res => {
-        this.recordUtils.parseResForErrors(res);
-
-        // update the records collectionId
-        this.updateRecordsCollectionId(this.collection._id);
-
+      const res = await this.factoryService.editCollection(collection);
+      if (!res || res.ok !== 1) {
+        alert('Failed to create collection.');
+      } else {
         this.loadingScreenService.setLoadingState(false, 'main');
         this.router.navigate(['mines', this.collection._master, 'collections', this.collection._id, 'detail']);
-      });
+      }
     } else {
       collection['_master'] = this.route.snapshot.paramMap.get('mineId');
       collection['project'] = this.route.snapshot.paramMap.get('mineId');
 
-      this.factoryService.createCollection(collection).subscribe(async (res: any) => {
-        this.recordUtils.parseResForErrors(res);
-
-        const createdCollection = res && res.length && res[0] && res[0].length && res[0][0] && res[0][0].object;
+      const res = await this.factoryService.createCollection(collection);
+      if (!res || res.ok !== 1) {
+        alert('Failed to create collection.');
+      } else {
+        const createdCollection = res.ops[0];
```

Comment:
creating collection and saving (with no added records) gets stuck on the spinner and throws res.ops is undefined
but the collection does get created